### DISCUSSION TITLE
Ondsel Vision

### DISCUSSION THREAD

**Discussion by sliptonic** [Ondsel-Development/FreeCAD#1] [Blog Topics]
We're a brand new company launched in early January 2023.  We're an Public Benefits Corporation building around the FreeCAD open-source software project.

Our goals include improving FreeCAD for everyone, improving the user interface and stability, and adding collaboration features that are difficult or impossible for the open-source community to build.

This discussion is created specifically to solicit feedback about our future and what you'd like to see from Ondsel and FreeCAD.

**Comment by szimszon** [comment]
Hi,

I love using FreeCAD and I would like to see toponaming fix in the first place ;) something like in [RealThunder's fork](https://github.com/realthunder/FreeCAD).

Great software please keep developing it!

Cheers

**Reply by ElPatoNoEsTu** [reply]
Hi, 

Same here -  I also like to see the TNP  adderessed first  ( migrating RT'S work to master ) .
I also think, that "Single Body" pradigma in PD is an atrificial restriction wich does not make any sense.  

Cheers, Stefan

**Reply by sliptonic** [reply]
Toponaming is going to be a big part of our lives.  No doubt about it.  
This subject got LOTS of discussion at Fosdem last weekend.  The great news is that we made progress on a plan to address it.
https://forum.freecad.org/viewtopic.php?t=75768&start=20#p658275

**Reply by szimszon** [reply]
Just curious  were also the previous efforts discussed? Like this: [first version of Toponaming is ready for testing](https://forum.freecad.org/viewtopic.php?t=73017&start=70) . Simple yes or no would fit. ;) I hope the answer will be yes. I waiting for years to jump back to vanilla version.

Cheers.

**Reply by sliptonic** [reply]
Yes.  We discussed it (a lot).  The post you linked to refers to the work done in [the toponaming branch](https://github.com/FreeCAD/FreeCAD/tree/development/toponaming).   They did some really great work to clean things up.  But here's the problem:  The branch represents 95 commits forward with toponaming but is almost 2500 commits behind master!  Before it could be merged, it would have to be rebased on master and that's another huge effort.  By the time the rebase is done, it will be hundreds or thousands of commits behind again.  You run in place and never catch up.  That's why I think the current approach is problematic.    

Instead of trying to get the whole thing ready to merge at once, we need to carve out pieces that can be merged to master immediately without breaking it.  Those first merges may not add anything useful but we can build on them without eternally rebasing.  It's going to take some time but it is achievable.

**Reply by szimszon** [reply]
Thank you for your answer. For me it puts all in the right context. This was the missing piece for me on that matter. So I look forward to see progress!

**Reply by nm17** [reply]
@sliptonic @szimszon How about we centralize the discussion and issue tracking in this Github Issue? https://github.com/Ondsel-Development/FreeCAD/issues/16

**Comment by adrianinsaval** [comment]
Given the comments above, I think it would be a good idea to clarify, is this a place where you want specific feature requests to be posted? (like toponaming and removal of single body paradigm requested above, which I suspect will be lost to the void) or is this a place where you want more general feedback, like "I want ondsel to work on improving functionality of the main freecad version" "I want ondsel to work on an enterprise support and training service" "I want ondsel to work on cloud services based around freecad"?

**Reply by sliptonic** [reply]
Good question.  Obviously, when you launch something new, the first thing you want is for people to pay attention.  Conversation - any conversation - is great. As we get our legs under us, of course, we want to focus the conversation in productive ways.  

For right now, I'm content to listen and see what people want to talk about and learn from them.  

If comments are things like,  "I hate the UI and this feature has a bug"  Then that tells us where we can help make FreeCAD better.  

If the comments are things like,  "I need to do XYZ in my business and I can't"  Then maybe we learn how our product can be better and where the commercial offering can produce revenue.

Both are valuable but for different reasons.

**Reply by howie-j** [reply]
I guess it's probably redundant to mention but if there is a need to get revenue/income relatively quickly, a proper bug bounty/pay for features program could potentially be successful, as with the recent b-spline work and as kipro-pcb does with KiCad.

**Reply by howie-j** [reply]
Also, Ondsel is really great news as far as i see, i wish you the best of luck and hope it will lead to a better FreeCAD for everyone😎

**Comment by adrianinsaval** [comment]
One possible method to get revenue is to have a modelling service: conversion of existing models in other formats to parametric FCStd, new models, fixing models, making technical drawings, etc. Ondsel could either be an intermediary to hire experts from the community (and takes a cut) or have hired modellers.

**Reply by sliptonic** [reply]
Interesting idea.  
Thinking about it from business perspective, it's tough to bootstrap because it's a 'marketplace'.  In other words, Ondsel would need to attract people to upload their models for conversion AND ALSO attract designers to do the conversion (or pay people to do it).  
Whenever you have to supply both sides of the transaction, it's hard to get going because neither side wants to participate until the other side is already there.  If you can get a marketplace going, they're great but bootstrap is challenging.

That said, this could definitely work as a follow-on business. Once FreeCAD designers are working in the tool, it would be easier to attract non-FreeCAD users to upload other kinds of models for conversion.

Thanks for the comment!

**Comment by Dods** [comment]
As controversial as it is, i think the best business model today is a subscription.
Not as in "we locked you in" but like, if you subscribe, you get professional help available via email.
I think that's the biggest issue with FreeCAD.
You can start using it, but very soon you will get issues adapting from the tool you previously used.
If, for example, there was an expert in FreeCAD available via email/ticket, i would gladly pay. Sometimes a video tutorial is just not cutting it.
I'm not interested in watching a 30min video. Just tell me how to make a hole. Or how to combine two bodies.
A ticket a dollar maybe. $10 sub/month gets you 10 tickets.
A subscription is not only about getting more. I would pay the $10 / month just to have a *option* of contacting someone and knowing my issue will get resolved.
FreeCAD UI is clunky.
Think Fusion360. When you select "combine", you pick a body to perform an operation, pick a body as a tool, select what you want (combine, cut) and click ok.
In FreeCAD i have absolutely no idea how to do that.
That means i'll have to continue to pay for Fusion until i watch all of those videos that explain how to do simple things.
I would gladly pay for an "onboarding" video or tutorial, or a "transition from" kind of document, where you go over all the simple things you need in order to get started.
Like, coming from Fusion, how to:
Create a rectangle
Extrude it
Create a sketch on a face of it
Create a hole
Make a thread
Create another simple body
Combine it with the first body
Now, i'm not a smart guy, but i never had any issues learning by doing while using Fusion.
A quick 2min search when getting stuck and that's it.
I have a lot of issues doing that in FreeCAD.
It is just not intuitive in any way, shape or form.
There is no simple documentation.
Leave the complex stuff for now.
Focus on the basics.
You have to educate your user base.
If a user downloads it and can't create a simple bracket, for example, that user won't be opening it again anytime soon.
Average user is not interested in fidgeting with options and preferences.
Maybe he will become interested later on, but certainly not in the first couple of days of usage.
Inconsistencies are awful.
When you are creating a circle, for example, you enter a dimension. It's a radius.
In the preferences i have selected diameter.
Why, when using dimension tool on a circle without a dimension, it listens for the option and uses diameter as it's default, but when creating the circle in the first place it uses radius?
There are tons of stuff like this, but i'll stop here because i'm not sure how interested you are in my blabber.

**Comment by valpackett** [comment]
Hi, here are some random thoughts from a random person:

- the most essential little feature from realthunder's fork is [storing projects as loose files instead of a zip archive](https://github.com/realthunder/FreeCAD/commit/1dfd9b93c9f) which is great for version control, e.g. I have an open hardware project with [the case design checked in](https://codeberg.org/valpackett/ergonice/src/branch/trunk/case) like that
- would be cool to have the [CfdOF](https://github.com/jaheyns/CfdOF) workbench included, and OpenFOAM & other other tools packaged inside the Flatpak (which yeah first there needs to be a Flatpak in general)
- on the business side, something that would entice me as a hobbyist is a cheap subscription that wouldn't necessarily include any collaborative stuff but would let me upload models to Lens, both public (searchable) and *unlisted* (no strict access control but not linked from public connections and `noindex` for search engines), with **parametric customization** similar to [Thingiverse's customizer](https://www.thingiverse.com/customizer)!
- … and since I mentioned CFD, running FEM/CFD/etc. solvers (and even meshers) in the "cloud" would also be a valuable service for the subscription! I think it would be super nice to have an as-easy-as-commercial integrated option for doing this, that doesn't hide anything, just gives you the choice to run the exact same FOSS solvers locally or remotely when you're trying to run a large project on a lower-end machine, or when you want to run a longer computation without having to have your machine powered on and using its resources for all that time.
